packages/world/ts/actions/callFrom.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  slice,
+  concat,
+  type WalletClient,
+  type Transport,
+  type Chain,
+  type Account,
+  type Hex,
+  type WalletActions,
+  type WriteContractReturnType,
+  type EncodeFunctionDataParameters,
+} from "viem";
+import { getAction, encodeFunctionData } from "viem/utils";
+import { writeContract } from "viem/actions";
+
+type CallFromParameters = {
+  worldAddress: Hex;
+  delegatorAddress: Hex;
+  worldFunctionToSystemFunction: (
+    worldFunctionSelector: Hex,
+  ) => Promise<{ systemId: Hex; systemFunctionSelector: Hex }>;
+};
+
+// By extending viem clients with this function after delegation, the delegation is automatically
+// applied to the World contract writes, meaning these writes are made on behalf of the delegator.
+export function callFrom<TChain extends Chain, TAccount extends Account>({
+  worldAddress,
+  delegatorAddress,
+  worldFunctionToSystemFunction,
+}: CallFromParameters): (
+  client: WalletClient<Transport, TChain, TAccount>,
+) => Pick<WalletActions<TChain, TAccount>, "writeContract"> {
```

Comment:
since we already have an assumption here about this being a world with `callFrom`, we could add an async check that the client account has a proper delegation set up for delegator address and warn if not

Reply:
makes sense!